plugins/georeplication/rest.go

```diff
@@ -367,3 +367,138 @@ func georepDeleteHandler(w http.ResponseWriter, r *http.Request) {
 
 	restutils.SendHTTPResponse(ctx, w, http.StatusOK, nil)
 }
+
+func georepStatusHandler(w http.ResponseWriter, r *http.Request) {
+	// Collect inputs from URL
+	p := mux.Vars(r)
+	masteridRaw := p["mastervolid"]
+	slaveidRaw := p["slavevolid"]
+
+	ctx := r.Context()
+	logger := gdctx.GetReqLogger(ctx)
+
+	// Validate UUID format of Master and Slave Volume ID
+	masterid, slaveid, err := validateMasterAndSlaveIDFormat(ctx, w, masteridRaw, slaveidRaw)
+	if err != nil {
+		return
+	}
+
+	geoSession, err := getSession(masterid.String(), slaveid.String())
+	if err != nil {
+		if _, ok := err.(*ErrGeorepSessionNotFound); !ok {
+			restutils.SendHTTPError(ctx, w, http.StatusInternalServerError, err.Error(), api.ErrCodeDefault)
+			return
+		}
+		restutils.SendHTTPResponse(ctx, w, http.StatusOK, []georepapi.GeorepSession{})
+		return
+	}
+
+	if geoSession.Status != georepapi.GeorepStatusStarted {
+		// Reach brick nodes only if Status is Started,
+		// else return just the monitor status
+		restutils.SendHTTPResponse(ctx, w, http.StatusOK, geoSession)
+		return
+	}
+
+	// Get Volume info, which is required to get the Bricks list
+	vol, e := volume.GetVolume(geoSession.MasterVol)
+	if e != nil {
+		restutils.SendHTTPError(ctx, w, http.StatusNotFound, errors.ErrVolNotFound.Error(), api.ErrCodeDefault)
+		return
+	}
+
+	// Status Transaction
+	txn := transaction.NewTxn(ctx)
+	defer txn.Cleanup()
+	lock, unlock, err := transaction.CreateLockSteps(masterid.String() + slaveid.String())
+	if err != nil {
+		restutils.SendHTTPError(ctx, w, http.StatusInternalServerError, err.Error(), api.ErrCodeDefault)
+		return
+	}
+
+	txn.Nodes = vol.Nodes()
+	txn.Steps = []*transaction.Step{
+		lock,
```

Comment:
Only to prevent other Volume operations which may lead to inconsistent status. (Volume deleted while getting Geo-rep status).

I will remove the lock, we can add it later if required.